scripts/scheduled-backup.sh

```diff
@@ -0,0 +1,198 @@
+#	    "$topdir/scripts/database-backup.sh" script, unless given
+#	    the "-n" command line option.  This writes a compressed dump
+#	    file to the "$topdir/db-backup" directory.
+#
+#	2.  Pull logs from the three Docker containers asv-main,
+#	    asv-db, and asv-rest.  These logs are stored in
+#	    "$topdir/log-backup".  Only the logging data produced since
+#	    the last time this script ran will be stored (empty log
+#	    backups are removed).  The timestamp on the symbolic link
+#	    "$topdir/backups/latest" (which is re-created at the end of
+#	    this script) is used to determine from when logs are to be
+#	    stored.
+#
+#	3.  Perform an incremental backup of the following directories
+#	    using rsync:
+#
+#		* "$topdir"/db-backup"	(the database backups)
+#		* "$topdir/log-backup"	(the log backups)
+#		* "asv-main:/uploads"	(the in-container upload directory)
+#
+#	The "$topdir" directory is the directory into which the mol-mod
+#	Github reposiory has been cloned.  This will be found via this
+#	script's location if the variable is left unset below.
+#
+#	Backups are written to "$topdir/backups/backup-{timestamp}",
+#	where "{timestamp}" is a timestamp on the "YYYYMMDD-HHMMSS"
+#	format.	 There will also be a symbolic link at "backups/latest",
+#	which will point to the most recent backup directory.
+#
+#	Backups are incremental.  Unchanged files are hard-linked in
+#	older backups and will not consume extra space.
+#
+#	This script is suitable to be run from crontab.	 Suggested
+#	crontab entry for twice-daily backups as 9 AM and 9 PM:
+#
+#		0 9,21 * * * /opt/mol-mod/scripts/scheduled-backup.sh
+#-----------------------------------------------------------------------
+
+usage () {
+	cat <<-END_USAGE
+	Usage:
+
+	    $0 -h
+
+	    $0 [-n] [-v]
+
+	Options:
+
+	    -h	displays this help text
+	    -n	do not dump database
+	    -v	be more verbose
+
+	END_USAGE
+}
+
+#-----------------------------------------------------------------------
+# Do command line parsing.
+#-----------------------------------------------------------------------
+
+be_verbose=false
+do_db_dump=true
+
+while getopts 'hnv' opt; do
+	case $opt in
+		h)
+			usage
+			exit
+			;;
+		n)
+			do_db_dump=false
+			;;
+		v)
+			be_verbose=true
+			;;
+		*)
+			echo 'Error in command line parsing' >&2
+			usage >&2
+			exit 1
+	esac
+done
+
+shift "$(( OPTIND - 1 ))"
+
+#-----------------------------------------------------------------------
+# Sanity checking our environment before starting.
+#-----------------------------------------------------------------------
+
+# Only run if the "asv-main" container is running.
+if [ "$( docker container inspect -f '{{ .State.Running }}' asv-main )" != 'true' ]
+then
+	echo 'Container "asv-main" not available'
+	exit 1
+fi >&2
+
+if ! command -v rsync >/dev/null 2>&1
+then
+	echo 'rsync is not available, or not in PATH' >&2
+	exit 1
+fi >&2
+
+unset topdir
+topdir=$( readlink -f "${topdir:-"$( dirname "$0" )/.."}" )
```

Comment:
Is `topdir` magical in some sh-incarnation? The use after unset confuses me.

Reply:
No it's not.  I will make that code less confusing right away. Thanks.

Reply:
Sorted out in a3d35b8